parser/parser.y

```diff
@@ -12648,17 +12656,18 @@ CreateRoleStmt:
 
 /* See http://dev.mysql.com/doc/refman/8.0/en/alter-user.html */
 AlterUserStmt:
-	"ALTER" "USER" IfExists UserSpecList RequireClauseOpt ConnectionOptions PasswordOrLockOptions CommentOrAttributeOption
+	"ALTER" "USER" IfExists UserSpecList RequireClauseOpt TokenRequireOpt ConnectionOptions PasswordOrLockOptions CommentOrAttributeOption
 	{
 		ret := &ast.AlterUserStmt{
 			IfExists:              $3.(bool),
 			Specs:                 $4.([]*ast.UserSpec),
 			TLSOptions:            $5.([]*ast.TLSOption),
-			ResourceOptions:       $6.([]*ast.ResourceOption),
-			PasswordOrLockOptions: $7.([]*ast.PasswordOrLockOption),
+			AuthTokenOptions:      $6.([]*ast.AuthTokenOption),
```

Comment:
I wonder if integrate `AuthTokenOptions` into `AuthOption` would be cleaner. That is `TokenRequireOpt` becomes part of `AuthOption:`. What do you think?

Reply:
I merged `TLSOption` and `AuthTokenOption` into `AuthTokenOrTLSOption`, PTAL :-)